src/Java.Interop.Tools.JavaTypeSystem/JavaModels/JavaMethodModel.cs

```diff
@@ -0,0 +1,188 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Java.Interop.Tools.JavaTypeSystem.Models
+{
+	public class JavaMethodModel : JavaMemberModel
+	{
+		public string Return { get; }
```

Comment:
Instead of the three properties `Return`, `ReturnGeneric`, and `ReturnJni`, should this instead be a `JavaTypeReference? ReturnType`, and add/permit a `JavaTypeReference(string rawJava, string genericJava, string jni)` constructor?